### ISSUE TITLE
how to update two connected models separatly?

### ISSUE THREAD

**Issue by sooyeonshin** [keras-team/keras#9722]
I have two CNN model, which connected like this.

> model1(input1, output)
> model2(model1.output, output2)

I want update each models separately every iteration and save weight separatly, but I don't know how to make this using keras model compiler and fit generator.

I coded like this for training operation

> model1.compile(optimizer=..., epochs=500)
> model2.compile(optimizer=..., epochs=500)

but I don't want model2 get model1 output which training operation is finished.
I want get model1 ouput at every iterations.

model1 is generate a image and model2 refines output of model1. 
I want to train with separate backpropation because each part has a different role.

> for every iteration
> model1(input1, output)
> update model1(loss(output))
> 
> model2(model1.output, output2)
> update model2(loss(output2))

**Comment by wenbotse** [comment]
@sooyeonshin may be you can try setting attribute layer.trainable of the layer to False which you don't want to train

**Comment by Ricocotam** [comment]
I think you should write your own learning loop : 
`
for epoch in range(nb_epoch):
    model1.fit(...)
    freeze(model1)  # function that freeze all layers
    model2.fit(...)
    unfreeze(model1)  # function that unfreeze all layers
`

I didn't try it but it should work
